src/test/ui/consts/std/poll.rs

```diff
@@ -0,0 +1,13 @@
+// run-pass
+
+use std::task::Poll;
+
+fn main() {
+    const POLL : Poll<usize> = Poll::Pending;
+
+    const IS_READY : bool = POLL.is_ready();
+    assert!(!IS_READY);
+
+    const IS_PENDING : bool = POLL.is_pending();
+    assert!(IS_PENDING);
+}
```

Comment:
Hm, why are we using a UI test for this, especially a `run-pass` one? I would think that these integrated tests are much costlier than standard `#[test]` tests, as well as much harder to debug. Linking a whole separate executable to check two const methods of stdlib is an overkill.

I would suggest testing similar things by unit or integrated `#[test]` tests in `core` or `std` itself. 

It might also be a good idea to sift through existing test suite for stdlib tests which *can* be moved to unit tests.

Reply:
Hm, yeah, we already have a bunch of tests like this...

I've opened https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/UI.20tests.20for.20stdlib for discussion, it might be that I am missing some context here of course.

Reply:
The conclusion is that we want to move such tests to `library/`: https://github.com/rust-lang/rust/issues/76268

@CDirkx if you could move some of the already existing const tests in a similar way, that would be awesome :)

Reply:
Sounds good, I'll have a look later today

Reply:
Tests are moved 👍